lookout/style/typos/preprocessing/metrics.py

```diff
@@ -0,0 +1,194 @@
+    the first k. Only the tokens, considered to be typoed, are taken into account
+    (correcting suggestions).
+    :param data: DataFrame which is indexed by "id" and has columns "typo", "corrupted".
+    :param suggestions: {id : [(candidate, correct_prob)]},
+                        candidates inside one suggestions list are
+                        sorted by correct_prob in a descending order.
+    :param k: Number of the first suggested corrections to check.
+    :return: Scores of correcting suggestions for score@k problem .
+    """
+    scores = Scores(0, 0, 0, 0)
+    for i in data.index:
+        if suggestions[i][0][0] != data.loc[i, COLUMNS["TOKEN"]]:
+            if data.loc[i, COLUMNS["TOKEN"]] != data.loc[i, COLUMNS["CORRECT_TOKEN"]]:
+                if data.loc[i, COLUMNS["CORRECT_TOKEN"]] in first_k_set(suggestions[i], k):
+                    scores.tp += 1
+                else:
+                    scores.fn += 1
+            else:
+                if data.loc[i, COLUMNS["CORRECT_TOKEN"]] in first_k_set(suggestions[i], k):
+                    scores.tn += 1
+                else:
+                    scores.fp += 1
+    return scores
+
+
+def accuracy(score: Scores) -> float:
+    """Calculate accuracy."""
+    return (score.tp + score.tn) / sum(list(score))
+
+
+def precision(score: Scores) -> float:
+    """Calculate precision."""
+    return score.tp / (score.tp + score.fp)
+
+
+def recall(score: Scores) -> float:
+    """Calculate recall."""
+    return score.tp / (score.tp + score.fn)
+
+
+def f1(score: Scores) -> float:
+    """Calculate f1 score."""
+    return 2 / (1 / precision(score) + 1 / recall(score))
+
+
+def print_score_metrics(score: Scores, file=None):
+    """Print score metrics."""
+    print(score, file=file)
+    print("Accuracy:", accuracy(score), file=file)
+    print("Precision:", precision(score), file=file)
+    print("Recall:", recall(score), file=file)
+    print("F1:", f1(score), file=file)
+
+
+def print_suggestion_results(data: pandas.DataFrame, suggestions: dict, path: str = None) -> None:
+    """Print suggestion results."""
+    file = None if not path else open(path, "w")
+    print("DETECTION SCORE\n", file=file)
+    print_score_metrics(detection_score(data, suggestions), file=file)
+    print("\nFIRST SUGGESTION SCORE\n", file=file)
+    print_score_metrics(score_at_k(data, suggestions, 1), file=file)
+    print("\nFIRST TWO SUGGESTIONS SCORE\n", file=file)
+    print_score_metrics(score_at_k(data, suggestions, 2), file=file)
+    print("\nFIRST THREE SUGGESTIONS SCORE\n", file=file)
+    print_score_metrics(score_at_k(data, suggestions, 3), file=file)
+
+
+def print_scores(data: pandas.DataFrame, suggestions: dict, path: str = None) -> None:
+    """Print scores for suggestions in an easy readable way."""
+    file = None if not path else open(path, "w")
+    print("{:15s}|{:15s}|{:15s}|{:15s}|{:15s}|{:15s}|{:15s}|{:15s}".format("METRICS",
```

Comment:
According to our coding conventions, we should use `%s` + please align along the single line